code/modules/mob/living/simple_animal/friendly/cat.dm

```diff
@@ -234,11 +234,20 @@
 		if("harm")
 			wuv(-1, M)
 
-/mob/living/simple_animal/pet/cat/proc/wuv(change, mob/M)
+/mob/living/simple_animal/pet/cat/proc/wuv(change, mob/living/M)
 	if(change)
 		if(change > 0)
 			if(M && stat != DEAD)
 				new /obj/effect/temp_visual/heart(loc)
+				if(M.has_trait(TRAIT_CAT_PERSON))

```

Comment:
uhm this is all backwards.

Send only a signal that a cat has been petted.

then let the thing that processes the signal figure out what trait the person has.

You're leaking the systems abstractions

Reply:
Unless I misunderstand moodlet code, wouldn't that require me to check every moodlet event to see if it's a certain type, and if so, if the traits were present?

Reply:
Currently, yes. You will have to edit moodlet code.

Reply:
@Qustinnus didn't I make you do this for something else, how did you implement it?

Reply:
I think my case was unique because everyone gets the same moodlet (table) except I modified it after if you are a catperson. Ill look into a different way of approaching this